hydra-node/src/Hydra/API/Server.hs

```diff
@@ -350,17 +353,29 @@ handleDraftCommitUtxo directChain tracer body reqMethod reqPaths respond = do
   case Aeson.eitherDecode' body :: Either String (DraftCommitTxRequest tx) of
     Left err ->
       respond $ responseLBS status400 [] (Aeson.encode $ Aeson.String $ pack err)
-    Right requestInput -> do
+    Right requestInput@DraftCommitTxRequest{utxos = draftUTxO'} -> do
       traceWith tracer $
         APIRestInputReceived
           { method = decodeUtf8 reqMethod
           , paths = reqPaths
           , requestInputBody = Just $ toJSON requestInput
           }
-
-      let userUtxo = utxos requestInput
-      eCommitTx <- draftTx userUtxo
-
+      let utxos =
+            ( \DraftUTxO{draftUTxO, draftScriptInfo} ->
+                (draftUTxO, draftScriptInfo)
+            )
+              <$> draftUTxO'
+      let
+        utxoInput =
+          ( \(utxo, maybeScriptInfo) ->
+              case maybeScriptInfo of
+                Just ScriptInfo{redeemer, datum, plutusV2Script} ->
+                  (utxo, Just (ScriptDatumForTxIn datum, redeemer, plutusV2Script))
+                Nothing ->
+                  (utxo, Nothing)
+          )
+            <$> utxos
```

Comment:
If we use different types in the API (makes sense because of json instances), do create a conversion function next to `DraftCommitTxRequest` along the lines of `convertDraftUTxO :: [DraftUTxO] -> [(TxIn, TxOut CtxUTxo, ...)]` (depending on the type used by `draftTx`)